app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -421,6 +431,41 @@ public void onStopTrackingTouch(SeekBar seekBar) {
             }
         });
 
+        // set zoom width of barcode
+        barcodeWidthScaler = binding.barcodeWidthScaler;
+        zoomWidthText = binding.zoomWidthText;
+        barcodeWidthScaler.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
+            @Override
+            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+                if (!fromUser) {
+                    Log.d(TAG, "non user triggered onProgressChanged, ignoring, progress is " + progress);
+                    return;
+                }
+                Log.d(TAG, "Progress is " + progress);
+                Log.d(TAG, "Max is " + barcodeWidthScaler.getMax());
+                float scale = (float) progress / (float) barcodeWidthScaler.getMax();
+                Log.d(TAG, "Scaling to " + scale);
+
+                loyaltyCard.zoomWidth = progress;
+                DBHelper.updateLoyaltyCardZoomWidth(database, loyaltyCardId, loyaltyCard.zoomWidth);
+
+                mainImage.getLayoutParams().width = mainLayout.getWidth() * loyaltyCard.zoomWidth / 100;
+                mainImage.requestLayout();
+
+                drawMainImage(mainImageIndex, true, isFullscreen);
+            }
+
+            @Override
+            public void onStartTrackingTouch(SeekBar seekBar) {
+
+            }
+
+            @Override
+            public void onStopTrackingTouch(SeekBar seekBar) {
+
+            }
+        });
+
```

Comment:
This seems like a copy-paste of the `barcodeScaler` function, I'd much rather see this turned into a single `SeekBar.OnSeekBarChangeListener` that both functions share.